x-pack/platform/plugins/shared/fleet/server/services/agent_policies/full_agent_policy.ts

```diff
@@ -323,12 +323,52 @@ export async function getFullAgentPolicy(
 }
 
 export function generateFleetConfig(
+  agentPolicy: AgentPolicy,
   fleetServerHosts: FleetServerHost,
-  proxies: FleetProxy[]
+  proxies: FleetProxy[],
+  outputs: Output[]
 ): FullAgentPolicy['fleet'] {
   const config: FullAgentPolicy['fleet'] = {
     hosts: fleetServerHosts.host_urls,
   };
+
+  // generating the ssl configs for checking into Fleet
+  // These are set in ES or remote ES outputs and correspond to --certificate-authorities, --elastic-agent-cert and --elastic-agent-cert-key cli options
+  const output = agentPolicy?.data_output_id
```

Comment:
I'm wondering if ssl config from the `monitoring_output_id` is not needed here?

Reply:
I'm not sure about the `monitoring_output_id`, I also wondered if I should add it here. I can add it and make sure to test it during end to end testing.

Reply:
Though I'm not sure how should it work if both `data_output_id` and `monitoring_output_id` has ssl config, we can't copy both to the same `fleet.ssl` config.